src/protocols/aeternity/helpers/actions.ts

```diff
@@ -0,0 +1,127 @@
+import { useStore } from 'vuex';
+import JsonBig from '@/lib/json-big';
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+import { ITransaction } from '@/types';
+import { fetchJson } from '@/utils';
+import { useAeSdk } from '@/composables/aeSdk';
+import { useMiddleware } from '@/composables/middleware';
+import { useAeNetworkSettings } from '@/protocols/aeternity/composables';
+import { AeSdkSuperhero } from '@/protocols/aeternity/libs/AeSdkSuperhero';
+
+async function fetchTransactionsFromMiddleware(
+  address: Encoded.AccountAddress,
+  nextPageUrl: string | null,
+  limit: number,
+) {
+  const {
+    fetchFromMiddlewareCamelCased,
+  } = useMiddleware();
```

Comment:
The role for utilities is to be a small functions that does simple things. So the general flow is that larger (more complex) entities imports smaller ones. Not the opposite. So here smaller one is using bigger one - I consider this as antipattern.